Library/Homebrew/diagnostic.rb

```diff
@@ -416,7 +416,7 @@ def check_xcode_up_to_date
       end
 
       def check_for_osx_gcc_installer
-        return unless MacOS.version < "10.7" || MacOS::Xcode.version > "4.1"
+        return unless MacOS.version < "10.7" || (!MacOS::Xcode.version.nil? && MacOS::Xcode.version > "4.1")
```

Comment:
That's a good idea; the current version is a bit hard to read. I'll switch to one of those.